engine/src/main/junit-shared/org/operaton/bpm/engine/test/util/JobExecutorHelper.java

```diff
@@ -0,0 +1,320 @@
+  private static final int JOB_EXECUTOR_WAIT_MULTIPLIER = 2;
+  private static final int THREAD_POOL_ACTIVE_COUNT_ZERO = 0;
+
+  private JobExecutorHelper() {
+    // Private constructor to prevent instantiation
+  }
+
+  /**
+   * Waits for the job executor to process all jobs within the process engine's default configuration
+   * using the default maximum wait time. This method internally delegates to
+   * {@link #waitForJobExecutorToProcessAllJobs(long)} using a predefined timeout.
+   */
+  public static void waitForJobExecutorToProcessAllJobs() {
+    waitForJobExecutorToProcessAllJobs(JOBS_WAIT_TIMEOUT_MS);
+  }
+
+  /**
+   * Waits for the job executor to process all jobs within the process engine's default configuration
+   * for the specified maximum time. It retrieves the process engine configuration and delegates
+   * the waiting logic to an internal implementation.
+   *
+   * @param maxMillisToWait the maximum time in milliseconds to wait for all jobs to be processed
+   */
+  public static void waitForJobExecutorToProcessAllJobs(long maxMillisToWait) {
+    // Get the process engine instance (previously done before each test run)
+    ProcessEngine processEngine = getProcessEngine();
+
+    // Retrieve the job executor from the process engine configuration
+    ProcessEngineConfigurationImpl processEngineConfiguration = ((ProcessEngineImpl) processEngine).getProcessEngineConfiguration();
+    waitForJobExecutorToProcessAllJobs(processEngineConfiguration, maxMillisToWait);
+  }
+
+  /**
+   * Waits for the job executor to process all jobs within the process engine configuration
+   * for the specified maximum time. The method ensures that no jobs remain unprocessed within
+   * the given time frame by invoking an internal implementation with a default check interval.
+   *
+   * @param processEngineConfiguration the process engine configuration providing access to the job executor
+   *                                   and management service
+   * @param maxMillisToWait            the maximum time in milliseconds to wait for all jobs to be processed
+   */
+  public static void waitForJobExecutorToProcessAllJobs(ProcessEngineConfigurationImpl processEngineConfiguration,
+                                                        long maxMillisToWait) {
+    // Check interval configuration (deprecated and unused prior to migration)
+    waitForJobExecutorToProcessAllJobs(processEngineConfiguration, maxMillisToWait, CHECK_INTERVAL_MS);
+  }
+
+  /**
+   * Waits for the job executor to process all jobs within the process engine configuration,
+   * ensuring that no jobs remain unprocessed within the specified time. The method repeatedly
+   * checks for job completion at the given intervals until the maximum time to wait is reached.
+   * If the job executor is not active, it will be started. The method will throw an exception
+   * if the jobs are not processed within the given time.
```

Comment:
The fact that the JobExecutor is shutdown after the execution might be important to know here.
But: Should the executor may only shut down when it was not active before?